File: java/business-context/src/main/java/cz/filipklimes/diploma/framework/businessContext/expression/IsNotNull.java
package cz.filipklimes.diploma.framework.businessContext.expression;

import cz.filipklimes.diploma.framework.businessContext.weaver.BusinessOperationContext;

import java.io.*;
import java.util.*;

public class IsNotNull<T> extends UnaryOperator<Boolean, T> implements Serializable
{

    public IsNotNull(final Expression<T> argument)
    {
        super(argument);
    }

    @Override
    public Boolean interpret(final BusinessOperationContext context)
    {
        return getArgument().interpret(context) != null;
    }

    @Override
    public Map<String, String> getProperties()
    {
        return Collections.emptyMap();
    }

    @Override
    public String getName()
    {
        return "is-not-null";
    }

    @Override
    public void accept(final ExpressionVisitor visitor)
    {
        visitor.visit(this);
    }

    @Override
    public String toString()
    {
        return String.format("%s is not null", getArgument());
    }

}

File: nodejs/business-context/src/expression/IsNotNull.js
export default class IsNotNull {
    constructor(argument) {
        this.argument = argument
    }

    interpret(context) {
        return this.argument.interpret(context) != null
    }

    getName() {
        return 'is-not-null'
    }

    getArguments() {
        return [this.argument]
    }

    getProperties() {
        return {}
    }

    toString() {
        return this.argument.toString() + " is not null"
    }

    clone() {
        return new IsNotNull(this.argument.clone())
    }
}
